snap/README.md

```diff
@@ -0,0 +1,167 @@
+
+However for full security confinement, the snap should be installed on an Ubuntu 18.04 LTS or later (Desktop or Server), or a system running Ubuntu Core 18 or later.
+
+### Installing EdgeX Device SNMP as a snap
+
+The snap is published in the snap store at https://snapcraft.io/edgex-device-snmp. You can see the current revisions available for your machine's architecture by running the command:
+
+```
+$ snap info edgex-device-snmp
+```
+
+The latest stable version of the snap can be installed using:
+
+```
+$ sudo snap install edgex-device-snmp
+```
+
+A specific release of the snap can be installed from a dedicated channel. For example, to install the 2.1 (Jakarta) release:
+
+```
+$ sudo snap install edgex-device-snmp --channel=2.1
+```
+
+The latest development version of the snap can be installed using:
+
+```
+$ sudo snap install edgex-device-snmp --edge
+```
+
+**Note** - the snap has only been tested on Ubuntu Core, Desktop, and Server.
+
+## Snap configuration
+
+Device services implement a service dependency check on startup which ensures that all of the runtime dependencies of a particular service are met before the service transitions to active state.
+
+Snapd doesn't support orchestration between services in different snaps. It is therefore possible on a reboot for a device service to come up faster than all of the required services running in the main edgexfoundry snap. If this happens, it's possible that the device service repeatedly fails startup, and if it exceeds the systemd default limits, then it might be left in a failed state. This situation might be more likely on constrained hardware (e.g. RPi).
+
+This snap therefore implements a basic retry loop with a maximum duration and sleep interval. If the dependent services are not available, the service sleeps for the defined interval (default: 1s) and then tries again up to a maximum duration (default: 60s). These values can be overridden with the following commands:
+
+To change the maximum duration, use the following command:
+
+```
+$ sudo snap set edgex-device-snmp startup-duration=60
+```
+
+To change the interval between retries, use the following command:
+
+```
+$ sudo snap set edgex-device-snmp startup-interval=1
+```
+
+The service can then be started as follows. The "--enable" option ensures that as well as starting the service now, it will be automatically started on boot:
+
+```
+$ sudo snap start --enable edgex-device-snmp.device-snmp
+```
+
+### Using a content interface to set device configuration
+
+The `device-config` content interface allows another snap to seed this device snap with both a configuration file and one or more device profiles.
+
+To use, create a new snap with a directory containing the configuration and device profile files. Your snapcraft.yaml file then needs to define a slot with read access to the directory you are sharing.
+
+```
+slots:
+  device-config:
+    interface: content  
+    content: device-config
+    write: 
```

Comment:
The key should be `read`, not `write` as it is exposing the read-only `$SNAP/config` directory.

Reply:
Thank you for your re-review, I have updated this part. I also updated this part's word based on your another [PR](https://github.com/edgexfoundry/device-mqtt-go/pull/311/files).